mautic/mautic
# DPMMA-2222 Email click heatmap

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ N ]
| New feature/enhancement? (use the a.x branch)      | [ Y ]
| Deprecations?                          | [ N ]
| BC breaks? (use the c.x branch)        | [ N ]
| Automated tests included?              | [ Y ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
This PR adds click heatmap visualization to emails in Mautic admin panel.

Preview:
![image](https://github.com/mautic/mautic/assets/8580942/18956d08-6c07-449a-b725-220b118e4fea)



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a new email with some links
3. Send the email to contacts and generate some clicks
4. Open the email view and select the Click heatmap from the dropdown next to the email controls:
![image](https://github.com/mautic/mautic/assets/8580942/3349f8ad-7cc8-4b31-901c-0d0f7ba7789a)
5. Test that everything is displayed correctly
